sdk/core/core/test/cmocka/test_az_json.c

```diff
@@ -51,10 +55,1074 @@ void test_json_token_number(void** state)
+    assert_true(c == 'r');
+    assert_true(_az_span_reader_read_json_string_char(&reader, &c) == AZ_OK);
+    assert_true(c == '\"');
+    assert_true(_az_span_reader_read_json_string_char(&reader, &c) == AZ_OK);
+    assert_true(c == 'u');
+    assert_true(_az_span_reader_read_json_string_char(&reader, &c) == AZ_OK);
+    assert_true(c == 'e');
+    assert_true(_az_span_reader_read_json_string_char(&reader, &c) == AZ_OK);
+    assert_true(c == '\t');
+    assert_true(_az_span_reader_read_json_string_char(&reader, &c) == AZ_ERROR_ITEM_NOT_FOUND);
+  }
+  {
+    az_span const s = AZ_SPAN_FROM_STR("\\uFf0F");
+    az_span reader = s;
+    uint32_t c = { 0 };
+    assert_true(_az_span_reader_read_json_string_char(&reader, &c) == AZ_OK);
+    assert_true(c == 0xFF0F);
+    assert_true(_az_span_reader_read_json_string_char(&reader, &c) == AZ_ERROR_ITEM_NOT_FOUND);
+  }
+  {
+    az_span const s = AZ_SPAN_FROM_STR("\\uFf0");
+    az_span reader = s;
+    uint32_t c;
+    assert_true(_az_span_reader_read_json_string_char(&reader, &c) == AZ_ERROR_EOF);
+  }
+}
+
+/** Json Value **/
+static void test_json_value(void** state)
+{
+  (void)state;
+
+  az_json_token const json_boolean = az_json_token_boolean(true);
+  az_json_token const json_number = az_json_token_number(-42.3);
+  az_json_token const json_string = az_json_token_string(AZ_SPAN_FROM_STR("Hello"));
+
+  // boolean from boolean
+  {
+    bool boolean_value = false;
+    assert_true(az_json_token_get_boolean(&json_boolean, &boolean_value) == AZ_OK);
+    assert_true(boolean_value);
+  }
+  // boolean from number
+  {
+    bool boolean_value = false;
+    assert_true(az_json_token_get_boolean(&json_number, &boolean_value) == AZ_ERROR_ITEM_NOT_FOUND);
+  }
+
+  // string from string
+  {
+    az_span string_value = { 0 };
+    assert_true(az_json_token_get_string(&json_string, &string_value) == AZ_OK);
+    assert_true(az_span_is_content_equal(string_value, AZ_SPAN_FROM_STR("Hello")));
+  }
+  // string from boolean
+  {
+    az_span string_value = { 0 };
+    assert_true(az_json_token_get_string(&json_boolean, &string_value) == AZ_ERROR_ITEM_NOT_FOUND);
+  }
+
+  // number from number
+  {
+    double number_value = 0.79;
+    uint64_t const* const number_value_bin_rep_view = (uint64_t*)&number_value;
+
+    assert_true(az_json_token_get_number(&json_number, &number_value) == AZ_OK);
+
+    double const expected_value = -42.3;
+    uint64_t const* const expected_value_bin_rep_view = (uint64_t const*)&expected_value;
+
+    assert_true(*number_value_bin_rep_view == *expected_value_bin_rep_view);
+  }
+  // number from string
+  {
+    double number_value = 0.79;
+    assert_true(az_json_token_get_number(&json_string, &number_value) == AZ_ERROR_ITEM_NOT_FOUND);
+  }
+}
+
+int test_az_json()
+{
+  const struct CMUnitTest tests[] = {
+    cmocka_unit_test(test_json_token_null),   cmocka_unit_test(test_json_token_boolean),
```

Comment:
Can we keep it one method per line, to be consistent with the rest?

Reply:
this is our auto-formating XD,  I think we can rename one test to be longer and that will converter it to one line

Reply:
In that case, it's fine. No need to artificially make test names longer :)